.github/actions/write-file/action.yml

```diff
@@ -13,19 +17,56 @@ inputs:
     description: "The name of the branch to commit to"
     required: true
     type: string
+  file_source:
+    description: "The file to be copied (incompatible with `content`)"
+    required: false
+    type: string
+  token:
+    description: "The token to use for authentication"
+    type: string
+  message:
+    description: "The commit message"
+    required: false
+    type: string
+  checkout_path:
+    description: "The path to checkout the repository to"
+    required: false
+    default: repo
+    type: string
 
 runs:
   using: composite
   steps:
     - name: Checkout repository
       uses: actions/checkout@v3
+      with:
+        repository: ${{ inputs.repository }}
+        ref: ${{ inputs.branch }}
+        path: ${{ inputs.checkout_path }}
+        token: ${{ inputs.token }}
 
     - name: Write file
       run: |
-        git config user.name github-actions[bot]
-        git config user.email github-actions[bot]@users.noreply.github.com
-        git reset --hard HEAD && git fetch origin ${{ inputs.branch }} && git checkout ${{ inputs.branch }} || exit 1
-        echo ${{ inputs.content }} > ${{ inputs.filepath }}
-        git add ${{ inputs.filepath }}
-        git commit -m "Write file: ${{ inputs.filepath }}" && git push -f
+        git config --global user.name github-actions[bot]
+        git config --global user.email github-actions[bot]@users.noreply.github.com
+
+        DEST="${{ inputs.checkout_path }}/${{ inputs.filepath }}"
+        mkdir -p "$(dirname \"$DEST\")"
+        if [ -n "${{ inputs.file_source }}" ]; then
+          cp ${{ inputs.file_source }} "$DEST"
+        elif [ -n "${{ inputs.content }}" ]; then
+          echo "${{ inputs.content }}" > "$DEST"
+        else
+          echo "No content or file_source specified"
+          exit 1
+        fi
```

Comment:
nitpick (non-blocking): you implicitly prioritize file source over content source in the logic here, but in the input list `content` comes first. To me it makes more sense that the "preferred" input is listed first. Otherwise it would be also be better to fail the workflow if both options are provided.

Reply:
Should it be preferred though? I actually forgot about handling multiple inputs being set, I will error-out instead.

Reply:
error-out is better yes

Reply:
you can go ahead and close out after that fix, my approval is still valid